dpdata/lammps/dump.py

```diff
@@ -197,7 +197,87 @@ def load_file(fname: FileType, begin=0, step=1):
                 buff.append(line)
 
 
-def system_data(lines, type_map=None, type_idx_zero=True, unwrap=False):
+def get_spin_keys(inputfile):
+    """
+    Read input file and get the keys for spin info in dump.
+
+    Parameters
+    ----------
+    inputfile : str
+        Path to the input file.
+
+    Returns
+    -------
+    list or None
+        List of spin info keys if found, None otherwise.
+    """
+    if inputfile is None or not os.path.isfile(inputfile):
```

Comment:
what is the situation where inputfile is not None but it does not exist?

Reply:
User pass a non-existent input file name.

Reply:
Why not throw an error?